src/lseq/lseq.rs

```diff
@@ -415,7 +437,9 @@ impl<V: Ord + Clone + Display, A: Actor + Display> LSeq<V, A> {
 
             // TODO: we may need a seed provided by the user to we get a deterministic result
             //let n = thread_rng().gen_range(reference_num - step, reference_num);
-            let n = reference_num - (step / 2) - 1;
+            // (drusu): no need to subtract here, we can always create larger numbers by adding a level,
+            //          so there's no need to shy away from hitting the max arity at this level
+            let n = reference_num - (step / 2); //  - 1;
```

Comment:
Not sure if I understand, but if you mean it's ok to use the max arity at any level then that brings issues depending on the strategy you are using at each level. A simple scenario (there are already a tests for these scenarios), let's say your arity at root is 4, and you already have a value with `Identifier([2])` and you want to insert between `Identifier([2])` and `END`, you cannot assign `Identifier([3])` if you are using Boundary- strategy, since next time you may get an insert between `Identifier([3])` and `END` and you neither can create `Identifier([4])`, nor respect the Boundary- strategy if you add a child to `Identifier([3])`. What I'm doing in these cases is creating a next level and using the max arity in the Identifier but never as the last number of it, so 0 and max arity are never the last number of any Identifier generated. E.g., in the scenario I just described you'd get `Identifier([2])`, `Identifier([3, 5])` instead. You can think of analogous cases for the Boundary+ when inserting at BEGIN.